src/reader/geometry_collection.rs

```diff
@@ -22,14 +22,17 @@ pub struct GeometryCollection<'a> {
 impl<'a> GeometryCollection<'a> {
     pub fn try_new(buf: &'a [u8], byte_order: Endianness, dim: Dimension) -> WkbResult<Self> {
         let mut offset = 0;
-        let has_srid = has_srid(buf, byte_order, offset);
+        let has_srid = has_srid(buf, byte_order, offset)?;
         if has_srid {
             offset += 4;
         }
 
         let mut reader = Cursor::new(buf);
         reader.set_position(HEADER_BYTES + offset);
-        let num_geometries = reader.read_u32(byte_order).unwrap().try_into().unwrap();
+        let num_geometries = reader
+            .read_u32(byte_order)?
+            .try_into()
+            .map_err(|e| WkbError::General(format!("Invalid number of geometries: {}", e)))?;
```

Comment:
I also wish the variants of WkbError were more informative. It's a bad habit of mine in early project development to have a "general" bucket.

That's something to add for the list of things for the next breaking release.

Reply:
It is hard to categorize various causes of WKB parsing failures, I believe that a general purpose error type with informative error messages should be good enough. GEOS and JTS also reports WKB parsing error using a single exception type `ParseException`.

Reply:
I don't think the calling use can do much except print the error text, so from that perspective I think General is fine.

Though, we might be able to eak out a little more perf by getting rid of `String` errors.

Reply:
We'd still be allocating the String - I don't expect Cow to be of much use here.